packages/dds/matrix/src/test/memory/sharedMatrix.spec.ts

```diff
@@ -0,0 +1,205 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { type IMemoryTestObject, benchmarkMemory } from "@fluid-tools/benchmark";
+import { MockFluidDataStoreRuntime } from "@fluidframework/test-runtime-utils/internal";
+
+import { SharedMatrix } from "../../index.js";
+import { matrixFactory, createLocalMatrix } from "../utils.js";
+
+describe("SharedMatrix memory usage", () => {
+	const matrixValue = "testValue";
+	const matrixSize = [10, 100, 1000, 10000];
+	const operationCounts = [10, 100, 1000, 10000];
+
+	beforeEach(async () => {});
+
+	afterEach(() => {});
+
+	for (const size of matrixSize) {
+		describe(`Size of ${size}*${size} SharedMatrix`, () => {
+			for (const count of operationCounts) {
+				benchmarkMemory(
+					new (class implements IMemoryTestObject {
+						readonly title = `Insert a column in the middle ${count} times`;
+						private localMatrix: SharedMatrix = matrixFactory.create(
+							new MockFluidDataStoreRuntime(),
+							"testLocalMatrix",
+						);
+
+						async run(): Promise<void> {
+							for (let i = 0; i < count; i++) {
+								this.localMatrix.insertCols(size / 2, 1);
+							}
+						}
+
+						beforeIteration(): void {
+							this.localMatrix = createLocalMatrix("testLocalMatrix", size, matrixValue);
+						}
+					})(),
+				);
+
+				benchmarkMemory(
+					new (class implements IMemoryTestObject {
+						readonly title = `Remove a column in the middle ${count} times`;
+						private localMatrix: SharedMatrix = matrixFactory.create(
+							new MockFluidDataStoreRuntime(),
+							"testLocalMatrix",
+						);
+
+						async run(): Promise<void> {
+							for (let i = 0; i < count; i++) {
+								this.localMatrix.removeCols(size / 2, 1);
+							}
+						}
+
+						beforeIteration(): void {
+							this.localMatrix = createLocalMatrix(
+								"testLocalMatrix",
+								size + count,
+								matrixValue,
+							);
+						}
+					})(),
+				);
+
+				benchmarkMemory(
+					new (class implements IMemoryTestObject {
+						readonly title = `Insert a row in the middle ${count} times`;
+						private localMatrix: SharedMatrix = matrixFactory.create(
+							new MockFluidDataStoreRuntime(),
+							"testLocalMatrix",
+						);
+
+						async run(): Promise<void> {
+							for (let i = 0; i < count; i++) {
+								this.localMatrix.insertRows(size / 2, 1);
```

Comment:
All of the locations where you're using `size / 2` should be updated to use `Math.floor(size / 2)` (or some other rounding utility). Otherwise it is possible to pass in non-integer values to code that only accepts integers.

The code currently works because all of our `size` values happen to be even. But a dev could change that and suddenly encounter very confusing errors.